idea/src/org/jetbrains/kotlin/idea/intentions/ConvertToForEachFunctionCallIntention.kt

```diff
@@ -35,15 +34,45 @@ class ConvertToForEachFunctionCallIntention : SelfTargetingIntention<KtForExpres
     override fun applyTo(element: KtForExpression, editor: Editor?) {
         val commentSaver = CommentSaver(element)
 
+        val labelName = element.getLabelName()
+
         val body = element.body!!
         val loopParameter = element.loopParameter!!
 
         val functionBodyArgument: Any = if (body is KtBlockExpression) body.contentRange() else body
 
-        val foreachExpression = KtPsiFactory(element).createExpressionByPattern(
+        val psiFactory = KtPsiFactory(element)
+        val foreachExpression = psiFactory.createExpressionByPattern(
                 "$0.forEach{$1->$2}", element.loopRange!!, loopParameter, functionBodyArgument)
-        val result = element.replace(foreachExpression)
+        val result = element.replace(foreachExpression) as KtElement
+
+        result.getContinuesWithLabel(labelName).forEach {
+            it.replace(psiFactory.createExpression("return@forEach"))
+        }
 
         commentSaver.restore(result)
     }
+
+    private fun KtElement.getContinuesWithLabel(labelName: String?): List<KtContinueExpression> {
+        val continueElements = ArrayList<KtContinueExpression>()
+
+        forEachDescendantOfType<KtContinueExpression>({ it !is KtLoopExpression }) {
```

Comment:
@yole 
Regarding your first comment: 
There is a certain logic behind the two loops. The first one never enters nested loops and collects all the unqualified `continue`s while the second one also enters nested loops. If I merged the two, how would I know whether to collect an unqualified `continue`?

Regarding your second comment:
Can you show me a code snippet that would be broken by the current implementation? In my understanding a `continue` inside a lambda/function/class would need to be inside a separate loop in which case the `it !is KtLoopExpression` check would prevent me from collecting them.